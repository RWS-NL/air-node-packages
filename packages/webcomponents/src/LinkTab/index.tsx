import React, { memo } from 'react';
import { Tab, TabProps } from '@mui/material';

import css from './LinkTab.scss';
import { Link } from 'react-router-dom';

export interface LinkTabProps {
  /** The label to show in the tab */
  label: string;
  /** The URL for this tab to navigate to */
  to: string;
  /**
   * Whether this link should go to an external site or not, using an `<a href=...>` component
   * @default false
   */
  external?: boolean;
  /**
   * Whether this link should open in a new tab or not
   * Only applies when external = true
   * @default false
   */
  openInNewTab?: boolean;

  /** Additional props to pass to each Tab component */
  TabProps?: TabProps;
}

export const LinkTab = memo(({ to, label, external = false, openInNewTab = false, ...props }: LinkTabProps) => {
  if (external) {
    return (
      <Tab
        {...props}
        component='a'
        href={to}
        label={label}
        target={openInNewTab ? '_blank' : '_self'}
        rel={openInNewTab ? 'noreferrer noopener' : undefined}
        classes={{ root: css.linkText, selected: css.selectedTab, textColorInherit: css.linkTextInherit }}
      />
    );
  }

  return (
    <Tab
      {...props}
      component={Link}
      to={to}
      label={label}
      classes={{ root: css.linkText, selected: css.selectedTab, textColorInherit: css.linkTextInherit }}
    />
  );
});
