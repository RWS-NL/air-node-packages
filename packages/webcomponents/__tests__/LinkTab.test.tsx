import { LinkTab } from '@src/LinkTab';
import { shallow } from 'enzyme';
import React from 'react';

describe('LinkTab Tests', function () {
  describe('Snapshot Testing', () => {
    test('Required Props', () => {
      const linkTab = shallow(<LinkTab to='/root' label='sample' />);
      expect(linkTab).toMatchSnapshot();
    });

    test('External Tab', () => {
      const linkTab = shallow(<LinkTab to='https://cgi.com' label='sample' external />);
      expect(linkTab).toMatchSnapshot();
    });

    test('External Tab with Open In New Tab', () => {
      const linkTab = shallow(<LinkTab to='https://rijkswaterstaat.nl' label='sample' external openInNewTab />);
      expect(linkTab).toMatchSnapshot();
    });
  });
});
