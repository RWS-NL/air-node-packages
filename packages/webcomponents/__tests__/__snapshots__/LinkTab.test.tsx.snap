// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkTab Tests Snapshot Testing External Tab 1`] = `
<ForwardRef(Tab)
  classes={
    Object {
      "selected": "selectedTab",
      "textColorInherit": "linkTextInherit",
      "wrapped": "linkText",
    }
  }
  component="a"
  href="https://cgi.com"
  label="sample"
  target="_self"
/>
`;

exports[`LinkTab Tests Snapshot Testing External Tab with Open In New Tab 1`] = `
<ForwardRef(Tab)
  classes={
    Object {
      "selected": "selectedTab",
      "textColorInherit": "linkTextInherit",
      "wrapped": "linkText",
    }
  }
  component="a"
  href="https://rijkswaterstaat.nl"
  label="sample"
  rel="noreferrer noopener"
  target="_blank"
/>
`;

exports[`LinkTab Tests Snapshot Testing Required Props 1`] = `
<ForwardRef(Tab)
  classes={
    Object {
      "selected": "selectedTab",
      "textColorInherit": "linkTextInherit",
      "wrapped": "linkText",
    }
  }
  component={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "propTypes": Object {
        "innerRef": [Function],
        "onClick": [Function],
        "replace": [Function],
        "target": [Function],
        "to": [Function],
      },
      "render": [Function],
    }
  }
  label="sample"
  to="/root"
/>
`;
